src/vcpkg/base/downloads.cpp

```diff
@@ -171,8 +165,8 @@ namespace vcpkg
         if (!multi_handle) Checks::unreachable(VCPKG_LINE_INFO);
 
         std::vector<int> ret(urls.size(), -1);
-        std::vector<CurlRequestPrivateData> private_data;
-        private_data.reserve(urls.size());
+        std::vector<WriteFilePointer> write_pointers;
+        write_pointers.reserve(urls.size());
```

Comment:
Note that this reserve is now load bearing to avoid the vector reallocating and changing the addresses of the `WriteFilePointer`s. Might be worth a comment.